piptools/writer.py

```diff
@@ -89,14 +90,15 @@ def _iter_lines(self, results, reverse_dependencies, primary_packages):
     def write(self, results, reverse_dependencies, primary_packages):
         with ExitStack() as stack:
             f = None
-            if not self.dry_run:
+            if not self.dry_run and self.dst_file != '-':
                 f = stack.enter_context(AtomicSaver(self.dst_file))
 
             for line in self._iter_lines(results, reverse_dependencies, primary_packages):
-                log.info(line)
```

Comment:
It does not use the logging module, but click: https://github.com/nvie/pip-tools/blob/d521d766113e69ea0a10017bb968d3b3bb242e5f/piptools/logging.py#L21-L22.